### ISSUE TITLE
JSP: Cannot use el expression(EL library conflicting issue)

### ISSUE THREAD

**Issue by daisuke-yoshimoto** [Activiti/Activiti#1685]
JSP: Cannot use el expression(EL library conflicting issue)

- [ ] 6.x
- [ ] develop

## Issue

* An error occurs when executing a jsp file containing an el expression.
* No error occurs when executing eL expressions in process execution.

### Related issues

* https://github.com/Activiti/Activiti/issues/1677
* https://community.alfresco.com/thread/232708-activiti-60javaxelelcontext-conflict
* https://github.com/Activiti/Activiti/issues/1038

### Cause

* The library 'juel' on which Activiti engine depends conflicts with the Expression Language 3.0 API(like jasper-el).
```
<dependency>
	<groupId>de.odysseus.juel</groupId>
	<artifactId>juel-impl</artifactId>
	<version>2.2.7</version>
</dependency>
<dependency>
	<groupId>de.odysseus.juel</groupId>
	<artifactId>juel-api</artifactId>
	<version>2.2.7</version>
</dependency>
<dependency>
	<groupId>de.odysseus.juel</groupId>
	<artifactId>juel-spi</artifactId>
	<version>2.2.7</version>
</dependency>
```
* juel-api-2.2.7.jar has the same interfaces as javax.el-api(Expression Language API Spec).
* Their interfaces's content is based on javax.el-api 2.x(Expression Language API Spec 2.x).
* So, juel-api-2.2.7.jar conflicts with the Expression Language 3.0 API(like jasper-el).

* On the other hand, Activiti-engine's java code depends on not only the interface of javax.el(Java EE spec) but also juel's implementation(juel-impl-2.2.7.jar).

### Why does this issue not reproduce in 5.x?

* Because Activiti engine embeds the juel implementation by repackaging.
* In 6.x and later versions, the repackaged juel implementation has been changed to juel's jar.
  * https://github.com/Activiti/Activiti/commit/a6ea01379d3117f55f434fe6d38afdb04f58aeb2#diff-ec32e4f09b82c973c1fb05cab8b41c26
---

### Plan1 for fix

It is an easy fix. But I don't know which this fix is perfect.
At least, all test case passes with this fix.

Remove the dependency for only juel-api and juel-spi that cause conflicts.
Remain the dependency for juel-impl.
```
<dependency>
	<groupId>de.odysseus.juel</groupId>
	<artifactId>juel-api</artifactId>
	<version>2.2.7</version>
</dependency>
<dependency>
	<groupId>de.odysseus.juel</groupId>
	<artifactId>juel-spi</artifactId>
	<version>2.2.7</version>
</dependency>
```

Instead, we should add the following dependency.
```
<!-- https://mvnrepository.com/artifact/javax.el/javax.el-api -->
<dependency>
    <groupId>javax.el</groupId>
    <artifactId>javax.el-api</artifactId>
    <version>2.2.5</version>
</dependency>
```

### Plan2 for fix

* Embed the juel implementation by repackaging like 5.x.
* I do not understand it, but it seems that such a correction is done in the project which forked us.
  * https://github.com/flowable/flowable-engine/commit/8c8a6018476b89afeffe492e999a0e0405d8923d#diff-20eb5f2f45779027a5aae9f51d6b18c7

**Comment by balsarori** [comment]
I'd rather go with the first option unless their is a good case for the second option.

**Comment by mteodori** [comment]
agree with @balsarori, we should check what happens with war projects that should have that dependency coming from the container so it might be necessary to set it as provided only in that case

**Comment by daisuke-yoshimoto** [comment]
@balsarori @mteodori 
Sorry for my late reply 😢 
Is it okay if I create a pull request to modify pom.xml for 6.x and develop each branches?

**Comment by mteodori** [comment]
thanks @daisuke-yoshimoto ok for me, not sure if we can/should switch to a javax.el 3.0 implementation instead where development is active provided we don't break any existing functionality, WDYT? also @balsarori

**Comment by balsarori** [comment]
I don't think we should use javax.el 3.0 especially for 6.x branch. It might be something to consider for develop branch.

**Comment by zhangzhd** [comment]
I operate according to plan one,but I don't add javax.el，it also works.

**Comment by salaboy** [comment]
@balsarori is there a decision on this one?

**Comment by cirodev** [comment]
I tried the first solution I'm using the version 7 of activiti integrated in spring

`<dependency>
			<groupId>org.activiti</groupId>
			<artifactId>activiti-spring-boot-starter</artifactId>
			<version>7.1.43</version>
			<exclusions>
				<exclusion>
					<groupId>commons-io</groupId>
					<artifactId>commons-io</artifactId>
				</exclusion>
				
				<exclusion>
					<groupId>de.odysseus.juel</groupId>
					<artifactId>juel-api</artifactId>
				</exclusion>
				<exclusion>
					<groupId>de.odysseus.juel</groupId>
					<artifactId>juel-spli</artifactId>
				</exclusion>
			</exclusions>
		</dependency>`

**Comment by cirodev** [comment]
I juste remove only juel-spi. Now it's work

**Comment by loraclaa** [comment]
Any update about this subject?We encounter the same question, and I deleted juel-spi, tomcat works.But I am not sure whether this lack jar will effect the main course of the Activiti6.0.0. Since this has  been marked as bug, which branch of 6.X will this be settled @daisuke-yoshimoto
